src/planner/constify_now.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <access/xact.h>
+#include <datatype/timestamp.h>
+#include <nodes/makefuncs.h>
+#include <optimizer/optimizer.h>
+#include <utils/fmgroids.h>
+
+#include "cache.h"
+#include "dimension.h"
+#include "hypertable.h"
+#include "hypertable_cache.h"
+#include "planner.h"
+
+/*
+ * This implements an optimization to allow now() expression to be
+ * used during plan time chunk exclusions. Since now() is stable it
+ * would not normally be considered for plan time chunk exclusion.
+ * To enable this behaviour we convert `column > now()` expressions
+ * into `column > const AND column > now()`. Assuming that times
+ * always moves forward this is save even for prepared statements.
+ *
+ * We consider the following expressions valid for this optimization:
+ * - Var > now()
+ * - Var >= now()
+ * - Var > now() - Interval
+ * - Var > now() + Interval
+ * - Var >= now() - Interval
+ * - Var >= now() + Interval
+ *
+ * Additionally Interval needs to be Const and not contain day or month
+ * components as those would be affected by timezone.
```

Comment:
```suggestion
 * Additionally Interval needs to be Const and not contain day or month
 * components as those would be affected by timezone, which can change
 * between executions of a prepared statement.
```

I'd add something like this.